python_modules/dagster/dagster/_core/storage/dagster_run.py

```diff
@@ -422,6 +424,18 @@ def tags_for_sensor(sensor) -> Mapping[str, str]:
     def tags_for_backfill_id(backfill_id: str) -> Mapping[str, str]:
         return {BACKFILL_ID_TAG: backfill_id}
 
+    @public
+    def get_event_records(self, instance: "DagsterInstance") -> Sequence["EventLogRecord"]:
```

Comment:
I thought about using `EventLogConnection` but it looks like the current version of this is sort of a dummy connection, it's just a wrapper for the resolved sequence: https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_core/storage/event_log/base.py#L36-L39

>Another option would be to return an Iterator and do batched fetching under the hood.

Seems like it would make sense to implement this inside the `EventLogConnection` class?

Reply:
see my comment in the main thread, I am now in favor of dropping this addition.